### PR TITLE
Adds NoOpTracerProvider test case for botocore instrumentation

### PR DESCRIPTION
**Pull request by shwejanraj**
# Description

Adds NoOpTracerProvider test cases for botocore instrumentation.

Fixes [#988](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/988)

# How Has This Been Tested?
tox -e test-instrumentation-botocore

# Does This PR Require a Core Repo Change?
No.

### CODE REVIEW

**Review by emdneto — COMMENTED** [review]
In https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2605 we updated `moto` package. You'll need to fix conflicts and update your tests accordingly

**Comment by shwejanraj** [comment]
@emdneto 
I have updated the test cases accordingly. Could you please take a moment to review it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `opentelemetry-instrumentation-asgi`, `opentelemetry-instrumentation-fastapi`, `opentelemetry-instrumentation-starlette` Use `tracer` and `meter` of originating components instead of one from `asgi` middleware
   ([#2580](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2580))
 
+### Added
+
+- `opentelemtry-instrumentation-botocore` Adds test case for NoOpTracerProvider ([#2586](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2586))
+
```
Comment (emdneto) [inline CHANGELOG.md:30]: We already have an section called Added, please move your changelog entry to there
```suggestion
```

**Review by lzchen — APPROVED** [review]
Please rebase and then allow maintainer edits so we can rebase for you.

**Comment by lzchen** [comment]
@shwejanraj 

Are you still working on this?

**Comment by Sunilwali679** [comment]
> @shwejanraj
> 
> Are you still working on this?

@lzchen . I am working on this and @shwejanraj  was working with me on this issue . I will rebase and update the PR . Thanks

**Comment by Sunilwali679** [comment]
@lzchen  https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2846  . created for the same

**Pull request closed without merging**